scriptler/discardOldBuilds.groovy

```diff
@@ -15,12 +15,20 @@
 // artifactDaysToKeep: If not -1 nor null, artifacts are only kept up to this days.
 // artifactNumToKeep: If not -1 nor null, only this number of builds have their artifacts kept.
 
-noLogRotation = hudson.model.Hudson.instance.items.findAll
-{job -> job.isBuildable() && job.logRotator==null}
-noLogRotation.each() { job ->
-    println job.name
-    if(!"true".equals(dryRun)){
-        job.logRotator = new hudson.tasks.LogRotator ( daysToKeep, numToKeep, artifactDaysToKeep, artifactNumToKeep)
-        println "$job.name fixed "
+import jenkins.model.Jenkins
+import hudson.tasks.LogRotator
+
+def setLogRotator(def instance) {
+    instance.items.findAll {job -> job.class.simpleName == 'Folder'}.each { setLogRotator(it) }
+    noLogRotation = instance.items.findAll {job -> job.isBuildable() && job.logRotator==null}
```

Comment:
Yes. _WorkflowJob_ and _AbstractProject_ (projects like Freestyle, Matrix, Maven, ...) are implementation of _Job_. 

Furthermore, the [hudson.model.Job|http://javadoc.jenkins-ci.org/hudson/model/Job.html] is the class that declare the **logRotator** and the **isBuildable()** methods.